test/unittests/execute_floating_point_test.cpp

```diff
@@ -69,4 +70,174 @@ TEST(execute_floating_point, f64_add)
+};
+template <>
+struct ConversionPairWasmTraits<double, int64_t>
+{
+    static constexpr auto opcode_name = "i64_trunc_f64_s";
+    static constexpr auto opcode = Instr::i64_trunc_f64_s;
+    static constexpr auto src_valtype = ValType::f64;
+    static constexpr auto dst_valtype = ValType::i64;
+};
+template <>
+struct ConversionPairWasmTraits<double, uint64_t>
+{
+    static constexpr auto opcode_name = "i64_trunc_f64_u";
+    static constexpr auto opcode = Instr::i64_trunc_f64_u;
+    static constexpr auto src_valtype = ValType::f64;
+    static constexpr auto dst_valtype = ValType::i64;
+};
+
+template <typename SrcT, typename DstT>
+struct ConversionPair : ConversionPairWasmTraits<SrcT, DstT>
+{
+    using src_type = SrcT;
+    using dst_type = DstT;
+};
+
+template <typename T>
+class execute_floating_point_conversions : public testing::Test
+{
+};
+
+struct ConversionName
+{
+    template <typename T>
+    static std::string GetName(int /*unused*/)
+    {
+        return T::opcode_name;
+    }
+};
+
+using pairs = testing::Types<ConversionPair<float, int32_t>, ConversionPair<float, uint32_t>,
+    ConversionPair<double, int32_t>, ConversionPair<double, uint32_t>,
+    ConversionPair<float, int64_t>, ConversionPair<float, uint64_t>,
+    ConversionPair<double, int64_t>, ConversionPair<double, uint64_t>>;
+TYPED_TEST_SUITE(execute_floating_point_conversions, pairs, ConversionName);
+
+TYPED_TEST(execute_floating_point_conversions, trunc)
+{
+    using src_type = typename TypeParam::src_type;
+    using dst_type = typename TypeParam::dst_type;
+    using src_limits = std::numeric_limits<src_type>;
+    using dst_limits = std::numeric_limits<dst_type>;
+
+    /* wat2wasm
+    (func (param f32) (result i32)
+      local.get 0
+      i32.trunc_f32_s
+    )
+    */
+    auto wasm = from_hex("0061736d0100000001060160017d017f030201000a070105002000a80b");
+
+    // Find and replace changeable values: types and the conversion instruction.
+    auto& param_type = wasm[wasm.find(static_cast<uint8_t>(ValType::f32))];
+    auto& result_type = wasm[wasm.find(static_cast<uint8_t>(ValType::i32))];
+    auto& opcode = wasm[wasm.find(static_cast<uint8_t>(Instr::i32_trunc_f32_s))];
+    param_type = static_cast<uint8_t>(TypeParam::src_valtype);
+    result_type = static_cast<uint8_t>(TypeParam::dst_valtype);
+    opcode = static_cast<uint8_t>(TypeParam::opcode);
+
+    auto instance = instantiate(parse(wasm));
+
+    {  // Max defined: something little smaller than 2^N.
+        const auto arg =
+            std::nextafter(std::pow(src_type{2}, src_type{dst_limits::digits}), src_type{0});
+        const auto expected = static_cast<dst_type>(arg);
+        const auto result = execute(*instance, 0, {arg});
+        EXPECT_EQ(result.value.template as<dst_type>(), expected);
+    }
+
+    {  // Min defined: -2^N or 0.
+        const auto arg =
+            dst_limits::is_signed ? -std::pow(src_type{2}, src_type{dst_limits::digits}) : 0;
```

Comment:
Maybe would make sense to test 0 for signed types, too, and move `-2^N` under `if (dst_limits::is_signed)` at the bottom of the test.

Reply:
Done.